api/chaos/serverexperimentation/v1/serverexperimentation.proto

```diff
@@ -56,3 +56,83 @@ enum FaultInjectionCluster {
   // Faults are injected to ingesss traffic by an upstream service(s).
   FAULTINJECTIONCLUSTER_UPSTREAM = 2;
 }
+
+message Config {
+  oneof fault {
+    option (validate.required) = true;
+
+    Abort abort = 1;
+    Latency latency = 2;
+  }
+}
+
+message Abort {
+  FaultTargeting fault_targeting = 1 [ (validate.rules).message.required = true ];
+  FaultPercentage fault_parcentage = 2 [ (validate.rules).message.required = true ];
+  AbortStatus abort_status = 3 [ (validate.rules).message.required = true ];
+}
+
+message Latency {
+  FaultTargeting fault_targeting = 1 [ (validate.rules).message.required = true ];
+  FaultPercentage fault_parcentage = 2 [ (validate.rules).message.required = true ];
+  LatencyDuration latency_duration = 3 [ (validate.rules).message.required = true ];
+}
+
+message FaultPercentage {
+  float percentage = 1 [ (validate.rules).float = {gt : 0.0, lte : 100.0} ];
+}
+
+message AbortStatus {
+  int32 http_status_code = 1 [ (validate.rules).int32 = {gt : 99, lt : 600} ];
+}
+
+message LatencyDuration {
+  int32 duration_ms = 1 [ (validate.rules).int32.gt = 0 ];
+}
+
+message FaultTargeting {
+  oneof enforcer {
+    option (validate.required) = true;
+
+    UpstreamEnforcing upstream = 1;
+    DownstreamEnforcing downstream = 2;
+  }
+}
+
+message UpstreamEnforcing {
+  oneof upstream {
+    option (validate.required) = true;
+
+    SingleCluster upstream_cluster = 1;
+    PartialSingleCluster upstream_partial_single_cluster = 2;
+  }
+
+  oneof downstream {
+    option (validate.required) = true;
+
+    SingleCluster downstream_cluster = 3;
+  }
+}
+
+message DownstreamEnforcing {
+  oneof upstream {
+    option (validate.required) = true;
+
+    SingleCluster upstream_cluster = 1;
+  }
+
+  oneof downstream {
+    option (validate.required) = true;
+
+    SingleCluster downstream_cluster = 2;
+  }
+}
+
+message SingleCluster {
+  string name = 1 [ (validate.rules).string = {min_bytes : 1} ];
+}
+
+message PartialSingleCluster {
+  string name = 1 [ (validate.rules).string = {min_bytes : 1} ];
+  float percentage = 2 [ (validate.rules).float = {gt : 0.0, lte : 100.0} ];
```

Comment:
Making this float gives the impression that we have float precision here, but I think realistically we'll have integer precision with a fixed denumerator. Might make sense to encode this here and for the other percent fields? Ie either let it specify both numerator/denumerator or a single uint32 for the numerator with a documented denumerator

Reply:
> a single uint32 for the numerator with a documented denumerator
sounds good, will update. I agree, realistically speaking there is not a lot of value in us supporting float precision